keel-echo/src/main/kotlin/com/netflix/spinnaker/keel/echo/UnhealthyNotificationListener.kt

```diff
@@ -1,37 +1,55 @@
 package com.netflix.spinnaker.keel.echo
 
+import com.netflix.spinnaker.config.UnhealthyNotificationConfig
 import com.netflix.spinnaker.keel.api.AccountAwareLocations
 import com.netflix.spinnaker.keel.api.Locatable
 import com.netflix.spinnaker.keel.api.Monikered
 import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.events.ClearNotificationEvent
 import com.netflix.spinnaker.keel.events.ResourceHealthEvent
 import com.netflix.spinnaker.keel.events.NotificationEvent
 import com.netflix.spinnaker.keel.notifications.ClusterViewParams
 import com.netflix.spinnaker.keel.notifications.Notification
 import com.netflix.spinnaker.keel.notifications.NotificationScope.RESOURCE
 import com.netflix.spinnaker.keel.notifications.NotificationType.UNHEALTHY_RESOURCE
+import com.netflix.spinnaker.keel.notifications.friendlyDuration
+import com.netflix.spinnaker.keel.persistence.UnhealthyRepository
 import com.netflix.spinnaker.keel.veto.unhealthy.UnsupportedResourceTypeException
 import org.springframework.beans.factory.annotation.Value
+import org.springframework.boot.context.properties.EnableConfigurationProperties
 import org.springframework.context.ApplicationEventPublisher
+import org.springframework.context.annotation.Configuration
 import org.springframework.context.event.EventListener
 import org.springframework.core.env.Environment
 import org.springframework.stereotype.Component
+import java.time.Duration
 
-
+@Configuration
+@EnableConfigurationProperties(UnhealthyNotificationConfig::class)
 @Component
 class UnhealthyNotificationListener(
+  private val config: UnhealthyNotificationConfig,
+  private val unhealthyRepository: UnhealthyRepository,
   private val publisher: ApplicationEventPublisher,
-  @Value("\${spinnaker.baseUrl}") private val spinnakerBaseUrl: String,
-  private val springEnv: Environment
+  private val springEnv: Environment,
+  @Value("\${spinnaker.baseUrl}") private val spinnakerBaseUrl: String
 ) {
 
   private val notificationsEnabled: Boolean
     get() = springEnv.getProperty("keel.notifications.unhealthy", Boolean::class.java, true)
 
   @EventListener(ResourceHealthEvent::class)
   fun onResourceHealthEvent(event: ResourceHealthEvent) {
-    if (notificationsEnabled && !event.healthy){
-      publisher.publishEvent(NotificationEvent(RESOURCE, event.resource.id, UNHEALTHY_RESOURCE, message(event.resource)))
+    if (notificationsEnabled) {
+      if (event.healthy) {
```

Comment:
it took me a minute to read `event.healthy` 😄  Is it possible to rename the property to `isResourceHealthy` or something like it?

Reply:
Is `isHealthy` okay? It's a resource health event, so the fact that it's a resource should be communicated by the name